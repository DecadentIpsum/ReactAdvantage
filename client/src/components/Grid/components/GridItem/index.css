.grid-item {
  padding: 1rem !important;
  border-right: 1px solid rgba(30, 30, 30, 0.2); }
  @media (max-width: 640px) {
    .grid-item {
      margin-bottom: 20px;
      border-right: 0;
      border-bottom: 1px solid rgba(30, 30, 30, 0.2);
      box-shadow: 0 1px 15px 1px rgba(81, 77, 92, 0.08); } }
  .grid-item:last-child {
    border-right: 0; }

.grid-item-title {
  color: #1E1E1E;
  font-size: 1.5rem;
  font-family: "Roboto", sans-serif;
  padding-bottom: 1rem;
  margin: 0; }
