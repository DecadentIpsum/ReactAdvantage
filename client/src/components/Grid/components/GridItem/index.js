import React, { Component } from 'react';
import './index.css';

export default class GridItem extends Component {
    render() {
        const {
            gridClassNames,
            title,
            ...restProps
        } = this.props;

        let classNames = ['grid-item'];

        if (gridClassNames) {
            classNames = classNames.concat(gridClassNames.split(' '));
        } else {
            classNames = classNames.concat('ui-g-12 ui-md-4 ui-lg-4'.split(' '));
        }

        return (
            <div
                className={classNames.join(' ')}
                {...restProps}
            >
                {title && <h3 className="grid-item-title">{title}</h3>}
                {this.props.children}
            </div>
        );
    }
}