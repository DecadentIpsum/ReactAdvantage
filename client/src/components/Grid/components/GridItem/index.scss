@import "global-styles/common.scss";

.grid-item {
    padding: 1rem !important;
    border-right: 1px solid rgba($blackShade, 0.2);

    @media(max-width: 640px) {
        margin-bottom: 20px;
        border-right: 0;
        border-bottom: 1px solid rgba($blackShade, 0.2);
        box-shadow: 0 1px 15px 1px rgba(81,77,92,.08);
    }

    &:last-child {
        border-right: 0;
    }
}

.grid-item-title {
    color: $blackShade;
    font-size: 1.5rem;
    font-family: $fonts;
    padding-bottom: 1rem;
    margin: 0;
}