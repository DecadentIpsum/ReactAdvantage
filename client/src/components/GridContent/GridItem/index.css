.grid-item {
  padding: 1rem !important;
  border-right: 1px solid rgba(30, 30, 30, 0.2); }
  @media (max-width: 640px) {
    .grid-item {
      margin-bottom: 20px;
      border-right: 0;
      border-bottom: 1px solid rgba(30, 30, 30, 0.2); } }
  .grid-item:last-child {
    border-right: 0; }
