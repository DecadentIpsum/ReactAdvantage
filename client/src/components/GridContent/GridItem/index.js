import React, { Component } from 'react';
import './index.css';

export default class GridItem extends Component {
    render(){
        return (
            <div className="ui-g-12 ui-md-4 ui-lg-4 grid-item">
                <h3 className="grid-item-title">{this.props.title}</h3>
                {this.props.item}
            </div>
        );
    }
}