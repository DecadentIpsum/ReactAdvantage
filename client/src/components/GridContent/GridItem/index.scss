.grid-item {
    padding: 1rem !important;
    border-right: 1px solid rgba(#1E1E1E, 0.2);

    @media(max-width: 640px) {
        margin-bottom: 20px;
        border-right: 0;
        border-bottom: 1px solid rgba(#1E1E1E, 0.2);
    }

    &:last-child {
        border-right: 0;
    }
}