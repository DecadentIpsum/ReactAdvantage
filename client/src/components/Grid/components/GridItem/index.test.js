import React from 'react';
import Enzyme, { shallow } from 'enzyme';

import GridItem from './index';

describe('it renders correctly', () => {
    test('GridItem renders correctly without additional class names', () => {
        const component = shallow(<GridItem />);

        expect(component).toMatchSnapshot();
    });

    test('GridItem renders correctly with additional class names', () => {
        const component = shallow(<GridItem gridClassNames="ui-lg-3" />);

        expect(component).toMatchSnapshot();
    });
});
