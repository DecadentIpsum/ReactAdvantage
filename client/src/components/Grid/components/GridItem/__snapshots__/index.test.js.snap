// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly GridItem render correctly with additional class names 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GridItem
    gridClassNames="ui-lg-3"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        undefined,
      ],
      "className": "grid-item ui-lg-3",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          undefined,
        ],
        "className": "grid-item ui-lg-3",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`it renders correctly GridItem render correctly without additional class names 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GridItem />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        undefined,
      ],
      "className": "grid-item ui-g-12 ui-md-4 ui-lg-4",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          undefined,
        ],
        "className": "grid-item ui-g-12 ui-md-4 ui-lg-4",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`it renders correctly GridItem renders correctly with additional class names 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GridItem
    gridClassNames="ui-lg-3"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        undefined,
      ],
      "className": "grid-item ui-lg-3",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          undefined,
        ],
        "className": "grid-item ui-lg-3",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`it renders correctly GridItem renders correctly without additional class names 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GridItem />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        undefined,
      ],
      "className": "grid-item ui-g-12 ui-md-4 ui-lg-4",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          undefined,
        ],
        "className": "grid-item ui-g-12 ui-md-4 ui-lg-4",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
